File: src/main/java/com/placement/portal/service/InterviewService.java
package com.placement.portal.service;

import java.util.List;

import com.placement.portal.dto.InterviewRequestDTO;
import com.placement.portal.dto.InterviewResponseDTO;

public interface InterviewService {

    InterviewResponseDTO scheduleInterview(
            Long applicationId,
            InterviewRequestDTO requestDTO);

    InterviewResponseDTO getInterviewByApplication(
            Long applicationId);

    boolean interviewExists(Long applicationId);

    List<InterviewResponseDTO> getAllInterviews();

    InterviewResponseDTO updateInterview(
            Long interviewId,
            InterviewRequestDTO requestDTO);

    void deleteInterview(Long interviewId);

    // NEW
    List<InterviewResponseDTO> getMyInterviews(String studentEmail);

}

File: frontend/src/services/InterviewService.js
import api from "../api/api";

export const scheduleInterview = (applicationId, data) => {
    return api.post(`/interviews/${applicationId}`, data);
};

export const getInterview = (applicationId) => {
    return api.get(`/interviews/application/${applicationId}`);
};

export const interviewExists = (applicationId) => {
    return api.get(`/interviews/exists/${applicationId}`);
};

export const getAllInterviews = () => {
    return api.get("/interviews");
};

export const updateInterview = (interviewId, data) => {
    return api.put(`/interviews/${interviewId}`, data);
};

export const deleteInterview = (interviewId) => {
    return api.delete(`/interviews/${interviewId}`);
};